SimpleNotePad.java

```diff
@@ -0,0 +1,225 @@
+ * 	readable.
+ * -Removed the println in the paste block. 
+ * -Moved some of the code from the if-else block in the actionPerformed method
+ *  to separate methods to make it more readable.
+ * -Removed unused code from paste function
+ * 
+ * -Added new features in the assignment spec
+ * 
+ */
+
+public class SimpleNotePad extends JFrame implements ActionListener{
+	private JMenuBar menuBar;
+	private JMenu fileMenu, editMenu, openRecentFile;
+	private JTextPane d = new JTextPane();
+	private JMenuItem newFile, saveFile, printFile, openFile;
+	private JMenuItem copy, paste, simpleReplace;
+	private JMenuItem[] recent = new JMenuItem[5];
+	private File[] recentFile = {null, null, null, null, null};
+	
+    public SimpleNotePad() {
+        setTitle("A Simple Notepad Tool");
+        menuBar = new JMenuBar();
+        initFileMenu();
+        initEditMenu();
+        menuBar.add(fileMenu);
+        menuBar.add(editMenu);
+        setJMenuBar(menuBar);
+        add(new JScrollPane(d));
+        setPreferredSize(new Dimension(600,600));
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setVisible(true);
+        pack();
+    }
+    
+    private void initFileMenu() {
+    	fileMenu = new JMenu("File");
+    	openRecentFile = new JMenu("Open Recent File");
+    	
+    	openFile = new JMenuItem("Open File");
+    	newFile = new JMenuItem("New File");
+    	saveFile = new JMenuItem("Save File");
+    	printFile = new JMenuItem("Print File");
+    	
+    	fileMenu.add(openFile);
+    	fileMenu.addSeparator();
+    	fileMenu.add(newFile);
+        fileMenu.addSeparator();
+        fileMenu.add(saveFile);
+        fileMenu.addSeparator();
+        fileMenu.add(printFile);
+        fileMenu.addSeparator();
+        fileMenu.add(openRecentFile);
+    	
+        openFile.addActionListener(this);
+        openFile.setActionCommand("open");
+        newFile.addActionListener(this);
+        newFile.setActionCommand("new");
+        saveFile.addActionListener(this);
+        saveFile.setActionCommand("save");
+        printFile.addActionListener(this);
+        printFile.setActionCommand("print");
+        openRecentFile.addActionListener(this);
+        openRecentFile.setActionCommand("open_recent");
+    }
+    
+    private void initEditMenu() {
+    	editMenu = new JMenu("Edit");
+    	simpleReplace = new JMenuItem("Replace");
+    	copy = new JMenuItem("Copy");
+    	paste = new JMenuItem("Paste");
+    	
+    	editMenu.add(simpleReplace);
+    	editMenu.add(copy);
+    	editMenu.add(paste);
+    	
+    	simpleReplace.addActionListener(this);
+    	simpleReplace.setActionCommand("simple_replace");
+    	copy.addActionListener(this);
+    	copy.setActionCommand("copy");
+    	paste.addActionListener(this);
+    	paste.setActionCommand("paste");
+    	
+    }
+    
+    private void addFileToRecent(File file) {
+    	openRecentFile.removeAll();
+    	for (int i = 4; i > 0; i--) {

```

Comment:
```suggestion
    	for (int i = recent.length; i > 0; i--) {
```
Iterator value int i should be initialized to recent.length so that any changes to the maximum number of files in the recently opened menu are inherited where it is needed).